### DISCUSSION TITLE
Could not import PIL IMage

### DISCUSSION THREAD

**Discussion by Saupia** [teamalgoritma/community#476] [Q&A] [answered]
Mohon bantuan untuk error pada coding berikut:
Pillow sudah terinstall. 

`model %>% 
  compile(
    loss = "categorical_crossentropy",
    optimizer = optimizer_adam(lr = 0.01),
    metrics = "accuracy"
  )

# install.packages("pillow")
# Fit data into model
history <- model %>% 
  fit(
  # training data
  image_train,

  # training epochs
  steps_per_epoch = as.integer(train_samples / batch_size), 
  epochs = 30, 
  
  # validation data
  validation_data = image_validation,
  validation_steps = as.integer(valid_samples / batch_size)
)

plot(history)`

![image](https://user-images.githubusercontent.com/114889462/206708363-184df280-869a-43c6-b469-e073275b478d.png)

**Comment by nisa-basalamah** [comment]
Dear @Saupia  ,

Terima kasih atas pertanyaan yang disampaikan, pertanyaan Anda sudah kami terima. Kami akan mencoba membantu mencarikan referensi berkaitan dengan pertanyaan yang diberikan.

**Comment by divakartika** [comment, marked as answer]
Dear @Saupia 

Terima kasih telah menghubungi kami. Dapat dikonfirmasi apakah proyek ini menggunakan Image Data Generator? Jika demikian maka proses fitting bukan menggunakan `fit()`, melainkan menggunakan `fit_generator()`. Dapat dicoba terlebih dahulu, apakah masih error?

**Reply by Saupia** [reply]
Halo Mbak Diva, 
Ini masih error. 
![image](https://user-images.githubusercontent.com/114889462/207022163-a5ee2de0-43c2-4d6d-ad40-04491db3e13d.png)

**Reply by divakartika** [reply]
Baik Pak Reinhard, kemungkinan terjadi error karena R belum terkoneksi dengan Python dengan baik sehingga library pillow tidak terdeteksi. Dapat dicoba terlebih dahulu untuk menggunakan `use_condaenv("nama_environment")` setelah import library-library yang digunakan. 

Jika masih terdapat error yang sama, dapat dicoba untuk run `reticulate::py_install("pillow",env=nama_environment)` sesuai referensi [berikut](https://stackoverflow.com/questions/59238504/tensorflow-with-r-and-anaconda-error-could-not-import-pil-image-the-use-of).

Boleh juga dibantu untuk melampirkan Rmd ke email mentor@algorit.ma untuk kami inspeksi code nya, terima kasih.
